CMakeLists.txt

```diff
@@ -23,6 +23,9 @@ endif(POLICY CMP0004)
 if(POLICY CMP0016)
   cmake_policy(SET CMP0016 OLD)
 endif(POLICY CMP0016)
+if(POLICY CMP0058) 
+  cmake_policy(SET CMP0058 NEW) 
+endif()
```

Comment:
I don't really know if we want this policy set.
As all it does is to suppress the warning of missing dependencies.
Maybe leave it unset in an developer/debug-build?

On the note of that, I tried to set it to `old` and ended up in an endless loop of cmake reconfigure.
This may also happen with an older version of `cmake` instead of an error.

Reply:
This policy has no effect in most cases, we still get warnings if dependencies could not be resolved: See here: https://cmake.org/cmake/help/v3.3/policy/CMP0058.html

> This policy has no effect on generators other than Ninja.

So, yes we want to set this policy as the users should only see actual warnings and not fake ones.

The gtest issue is not really related to ninja, but to cotire and a messy system setup. If you like, open an issue at the [cotire repo](https://github.com/sakra/cotire/issues).

IMO we can merge this PR as it does not even affect the unix make build-system. The ninja-build also should work for most people and if not they can switch back to make. If it makes you sleep better, add a message that ninja builds are experimental.

Reply:
I don't know if it is really a fake warning, as it is not displayed if `cotire` isn't included.